CONTRIBUTING.md

```diff
@@ -26,17 +26,17 @@ Code contributions via pull requests are much appreciated. Before sending us a p
 
 1. You are working against the latest source on the *master* branch.
 2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
-3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+3. You open an issue to discuss any significant work before implementing it. Major new features, submodules, and model contributions need to fit into our overall design philosophy and our user's interests. We also need to evaluate if the contribution is worth the maintenance overhead.
 
 To send us a pull request, please:
 
-1. Fork the repository.
+1. Fork the repository and clone the source code to your local machine.
```

Comment:
Should we add pointers to the official GitHub docs on [forking & cloning](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo)?

Reply:
great idea, added